src/main/java/com/aws/greengrass/shadowmanager/sync/model/CloudUpdateSyncRequest.java

```diff
@@ -85,98 +85,86 @@ public void merge(CloudUpdateSyncRequest other) {
     @Override
     public void execute(SyncContext context) throws RetryableException, SkipSyncRequestException,
             ConflictException, UnknownShadowException, InterruptedException {
-        // synchronizing access to shadow sync information because it is accessed by multiple threads.
-        // In this method, shadow sync information is accessed within context of the sync thread.
-        // The other place is `isUpdateNecessary(SyncContext)`, which runs on the IPC thread.
-        //
-        // Other types of sync requests don't need this lock because their implementation of
-        // `isUpdateNecessary(SyncContext)` does not access shadow sync information.
-        //
-        // The check for shadow sync information in this class' `isUpdateNecessary(SyncContext)`
-        // prevents unnecessary full syncs,
-        // (see https://github.com/aws-greengrass/aws-greengrass-shadow-manager/pull/106).
-        synchronized (context.getSynchronizeHelper().getThingShadowLock(this)) {
-            Optional<ShadowDocument> shadowDocument = context.getDao().getShadowThing(getThingName(), getShadowName());
+        Optional<ShadowDocument> shadowDocument = context.getDao().getShadowThing(getThingName(), getShadowName());
```

Comment:
removed the `synchronized` block since not we're not reading the sync info from multiple threads anymore